required/latex-lab/latex-lab-graphic.dtx

```diff
@@ -433,10 +433,36 @@
 %   \DocumentMetadata{tagging=on,debug=BBox}
 %  \end{verbatim}
 %  
+%  \subsection{Hooks}
+%  The three key definitions \texttt{alt}, \texttt{actualtext} and \texttt{artifact} used by
+%  \cs{includegraphics} contain hooks, named \texttt{Gin/alt},
+%  \texttt{Gin/actualtext}, and \texttt{Gin/artifact}. The first two take as argument
+%  the value of the key. The hooks are processed even if tagging is not activated.
+%  With them, it is for example possible to store the alternative text:
+%  \begin{verbatim}
+%  \AddToHookWithArguments{Gin/alt}{\gdef\myalttext{#1}}
+%  \includegraphics[alt=Hello World]{example-image-A}
+%  The alt text of the graphic was \myalttext.
+%  \end{verbatim}
+%  
+%  Please note
+%  \begin{itemize}
+%   \item The hooks are also executed if the keys are set with \verb+\setkeys{Gin}+.
+%   \item The arguments are given to the hooks without any processing or purifying. 
```

Comment:
why is that the case? Why not first purify and pass the resulting string to the hook? Might make more sense since that is what is going to be used. Or perhaps provide both to the hook: the purified version and the plan version, i.e. a hook with 2 arguments
```
         \tl_if_empty:nTF{#1}
           { \tl_clear:N\l_@@_graphic_alt_tl}
           { \tl_set:Ne\l_@@_graphic_alt_tl{\text_purify:n{#1}}}
        \exp_arg:Nnno  \hook_use:nnw {Gin/alt}{2} \l_@@_graphic_alt_tl {#1}       
```

Reply:
I didn't want to remove the option to handle the original argument. But passing both is naturally an option.

Reply:
I think it would be better. In most cases you want to present the same string as it is used by the algorithm but sometimes you may want to use what the user has typed.